test/streams.js

```diff
@@ -180,6 +180,27 @@ describe('a function returning a ReadableStream', function() {
     });
   });
 
+  describe('an http client requesting a stream as JSON', function() {
+    it('should close server stream on client disconnect',
+      function(done) {
+        const r = request(app)
+          .get('/StreamClass/createInfiniteStream')
+          .set('Accept', 'application/json')
+          // .set('Accept', 'text/event-stream')
+          .timeout(100)
+          .buffer(false);
+        r
+          .then(function() {
```

Comment:
Mixing promises (`.then()` API) with callbacks (`done` argument) is usually a recipe for a disaster. I'll take a look on how to rewrite this test to be either fully callback-based or fully promise-based.

BTW when I run this new test agains the current master, the test fails with the following error:

```
 1) a function returning a ReadableStream
       an http client requesting a stream as JSON
         should close server stream on client disconnect:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/bajtos/src/strong/remoting/test/streams.js)
```

I am not sure if that's intended?

Reply:
I managed to find a better way how to test this, will push the changes to your feature branch soon.